internal/proto/v1/hook.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+package v1;
+
+// Uploaded data
+message Upload {
+	// Unique integer identifier of the uploaded file
+	string id = 1;
+	// Total file size in bytes specified in the NewUpload call
+	int64 Size = 2; 
+	// Indicates whether the total file size is deferred until later
+	bool SizeIsDeferred = 3;
+	// Offset in bytes (zero-based)
+	int64 Offset = 4;
+	map<string, string> metaData = 5;
+	// Indicates that this is a partial upload which will later be used to form
+	// a final upload by concatenation. Partial uploads should not be processed
+	// when they are finished since they are only incomplete chunks of files.
+	bool isPartial = 6; 
+	// Indicates that this is a final upload
+	bool isFinal = 7; 
+	// If the upload is a final one (see IsFinal) this will be a non-empty
+	// ordered slice containing the ids of the uploads of which the final upload
+	// will consist after concatenation.
+	repeated string partialUploads = 8;
+	// Storage contains information about where the data storage saves the upload,
+	// for example a file path. The available values vary depending on what data
+	// store is used. This map may also be nil.
+	map <string, string> storage = 9;
+}
+
+message HTTPRequest {
+	// Method is the HTTP method, e.g. POST or PATCH
+	string method = 1;
+	// URI is the full HTTP request URI, e.g. /files/fooo
+	string uri = 2;
+	// RemoteAddr contains the network address that sent the request
+	string remoteAddr = 3; 
+}
+
+// Hook's data
+message Hook {
+	// Upload contains information about the upload that caused this hook
+	// to be fired.
+	Upload upload = 1;
+	// HTTPRequest contains details about the HTTP request that reached
+	// tusd.
+	HTTPRequest httpRequest = 2;
+}
+
+// Request data to send hook
+message SendRequest {
+  string Api = 1;
+  Hook hook = 2;
+}
+
+// Response that contains data for sended hook
+message SendResponse {
+  string Api = 1;
+  int64 statusCode = 2;
+}
+
+// The hook service definition.
+service HookService {
+  // Sends a hook
+  rpc Send (stream SendRequest) returns (stream SendResponse) {}
```

Comment:
What does the stream mean here? Could you explain it for a person that has never used gRPC before?

Reply:
@sagikazarmark  commented:
> I think it would make sense to play with the thought of making this a streaming RPC call. Given the content is actually an event stream (sort of).

- [gRPC concepts](https://grpc.io/docs/guides/concepts/)

The most general case is Bidirectional Streaming where a single gRPC call establishes a stream in which both the client and the server can send a stream of messages to each other. 
Reviewing my approach, I think it's not well implemented for bidirectional streaming. We have two options:

1. Make a simple RPC call using the same patter as HTTP hook. Create a connection for each hook.
2. Use bidirectional streaming to use the same connection for each hook.